eclipse-theia/theia
# Substitute ~ for home directory in file dialog

#### What it does

Fixes #9142 by adding support for '~' character in text input field of browser's file dialog. This MR also introduces some small refactors making factoritizing LocationRenderer and making it injectable to allow for cleaner dependency management.

#### How to test

- Build this branch for browser and run
- Open a file dialog (either save/open) and click on the 'folder' icon to switch to text input mode
- Begin typing a path that starts with ~, observe that autocompletion works as expected
- Navigate to a valid path that begins with ~ by pressing enter, observe it works as expected and there are no regressions

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
